core/performance_test/benchmark/Cabana_LinkedCellPerformance.cpp

```diff
@@ -0,0 +1,169 @@
+/****************************************************************************
+ * Copyright (c) 2018-2021 by the Cabana authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the Cabana library. Cabana is distributed under a   *
+ * BSD 3-clause license. For the licensing terms see the LICENSE file in    *
+ * the top-level directory.                                                 *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "Cabana_BenchmarkTimer.hpp"
+
+#include <Cabana_Core.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <numeric>
+#include <sstream>
+#include <string>
+#include <vector>
+
+//---------------------------------------------------------------------------//
+// Performance test.
+template <class Device>
+void performanceTest( std::ostream& stream, const std::string& test_prefix )
+{
+    // Declare problem sizes.
+    double min_dist = 1.0;
+    std::vector<int> problem_sizes = { 1000, 10000, 100000, 1000000 };
+    int num_problem_size = problem_sizes.size();
+    std::vector<double> x_min( num_problem_size );
+    std::vector<double> x_max( num_problem_size );
+
+    // Declare the number of cutoff ratios (directly related to neighbors per
+    // atom) to generate.
+    std::vector<double> cutoff_ratios = { 4.0, 8.0 };
+    int cutoff_ratios_size = cutoff_ratios.size();
+
+    // Declare the number of cell ratios (only used for Verlet) to generate.
+    std::vector<double> cell_ratios = { 1.0 };
+
+    // Number of runs in the test loops.
+    int num_run = 10;
+
+    // Define the aosoa.
+    using member_types = Cabana::MemberTypes<double[3]>;
+    using aosoa_type = Cabana::AoSoA<member_types, Device>;
+    std::vector<aosoa_type> aosoas( num_problem_size );
+
+    // Create aosoas.
+    for ( int p = 0; p < num_problem_size; ++p )
+    {
+        int num_p = problem_sizes[p];
+        // Define problem grid.
+        x_min[p] = 0.0;
+        x_max[p] = 1.3 * min_dist * std::pow( num_p, 1.0 / 3.0 );
+        aosoas[p].resize( num_p );
+        auto x = Cabana::slice<0>( aosoas[p], "position" );
+        Cabana::Benchmark::createParticles( x, x_min[p], x_max[p], min_dist );
+    }
+
+    // Loop over number of ratios (neighbors per particle).
+    for ( int c = 0; c < cutoff_ratios_size; ++c )
+    {
+        // Will need loop over cell ratio if more than one.
```

Comment:
Is this comment referring to the above for loop?

Reply:
This was added per your last comment